File: isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == "":
            return True
        if len(s) > len(t):
            return False
        sp, tp = 0, 0
        while sp < len(s):
            if t[tp] == s[sp]:
                sp += 1
            tp += 1
            if sp == len(s):
                return True
            if tp == len(t):
                return False

File: isSubsequence.js
var isSubsequence = function (s, t) {
  if (s == "") return true;
  if (s.length > t.length) return false;
  let sp = 0;
  let tp = 0;
  while (sp < s.length) {
    if (t[tp] != s[sp]) tp++;
    else if (t[tp] == s[sp]) {
      sp++;
      tp++;
    }
    if (sp == s.length) return true;
    if (tp == t.length) return false;
  }
};

module.exports = isSubsequence;
